### PR TITLE
feat: fetch prices

### PR DESCRIPTION
**Pull request by 0xArdy**
# 🤖 Linear

Closes FRA-32

### CODE REVIEW

File: src/types/Data.ts [hunk 1]
```diff
@@ -1,5 +1,11 @@
 import { Hex, Address } from 'viem';
 
+export interface CallResult {
+  result: bigint;
```
Comment (turtlemoji) [inline src/types/Data.ts:4]: is this type ok? is not BigInt?

File: src/utils/formatTokenData.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        address: tokenList[index].address,
+        decimals: tokenList[index].decimals,
+        balanceE18: result.toString(),
+        balance: formatUnits(result, tokenList[index].decimals),
+        balanceUSD: getUsdBalance(price, result.toString(), tokenList[index].decimals),
+        price: price,
+      };
+    });
+
+    const ethPrice = prices.coins[`${chainName}:${getConfig().DEFAULT_ETH_ADDRESS}`].price || 0;
```
Comment (turtlemoji) [inline src/utils/formatTokenData.ts:29]: duplicated logic, create const at the top

File: src/utils/getPrices.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { PriceData } from '~/types';
+import { fetchData } from './getContractAbi';
+
+export const getPricesUrl = (chainName: string, address: string[]) => {
+  const addresses = address.join(`,${chainName}:`);
+  return `https://coins.llama.fi/prices/current/${chainName}:${addresses}`;
```
Comment (turtlemoji) [inline src/utils/getPrices.ts:6]: should we always use this url? what happens if it fails?

Comment (0xArdy) [reply src/utils/getPrices.ts:6]: this function just returns the url as string. But if getPrices() fails it emit an error and returns an empty object

File: src/utils/formatTokenData.ts (hunk 2 above)
Comment (0xArdy) [reply src/utils/formatTokenData.ts]: done https://github.com/defi-wonderland/keep3r-framework-app/pull/3/commits/47777a741b618c82e489e947b96cd9e326054150

File: src/types/Data.ts (hunk 1 above)
Comment (0xArdy) [reply src/types/Data.ts:4]: yes its ok

**Review by turtlemoji — APPROVED** [review]

**Pull request merged**